go/main.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func printHello(w http.ResponseWriter) {
+	fmt.Fprintf(w, "Hello from Go\n")
+}
+
+func hello(w http.ResponseWriter, req *http.Request) {
+	printHello(w)
+}
+
+func proxyURL(orig *url.URL) *url.URL {
+	path := strings.TrimSuffix(orig.Path, "/")
+	idx := strings.LastIndex(path, "/")
+	return &url.URL{Scheme: "http", Host: path[idx+1:], Path: path[:idx]}
```

Comment:
I don't think it matters all that much, but I think the other implementations use the first segment after the `/proxy/` to be the host to proxy to, rather than the last one. This is definitely simpler to implement. ;)